internal/core/textlayout.rs

```diff
@@ -532,7 +557,7 @@ fn test_cursor_position() {
         vertical_alignment: TextVerticalAlignment::Top,
         wrap: TextWrap::WordWrap,
         overflow: TextOverflow::Clip,
-        single_line: false,
+        single_line: true,
```

Comment:
the test expects the cursor to stay on the first line i.e. all expected y-coordinates are 0. before, this test passed because the layout algo was too eager to truncate or elide any line even if wrapping was enabled.